File: 백준/Silver/1475. 방 번호/방 번호.c
#include <stdio.h>
#include <string.h>
int a[10];
char n[10];
int main(){
  scanf("%[^\n]s", &n);
  int len = strlen(n);
  for(int i = 0; i < len; i++){
    int cur = n[i] - '0';
    (cur == 9) ? a[6]++ : a[cur]++;
  }
  if(a[6]&1) a[6] += 1;
  a[6] /= 2;
  int max = -1;
  for(int i = 0; i < 10; i++){
    if(max < a[i]) max = a[i];
  }
  printf("%d", max);
}

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.*;
import java.util.Arrays;

public class Main {
  public static void main(String[] args) throws IOException{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    StringBuilder sb = new StringBuilder();
    String N = br.readLine();
    int[] freq = new int[9];
    for(char c : N.toCharArray()){
      int cur = c - '0';
      freq[(cur==9) ? 6 : cur]++;
    }
    freq[6] = (freq[6] + 1) / 2;
    sb.append(Arrays.stream(freq).max().getAsInt());
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }
}